packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -872,6 +955,17 @@ class _WidgetStateMapper<T> implements WidgetStateProperty<T> {
       );
     }
   }
+
+  @override
+  bool operator ==(Object other) {
+    return other is _WidgetStateMapper && mapEquals(map, other.map);
+  }
+
+  @override
+  int get hashCode => Object.hashAll(<Object?>[...map.keys, ...map.values]);
```

Comment:
Can't we use `const MapEquality()` 's `hashCode` utility here? Or is that not needed/overkill?

Reply:
ooh, I hadn't heard of `MapEquality()` till now! I wonder how its performance compares to the current getter 🤔

Reply:
I definitely feel safer using [MapEquality.hash](https://main-api.flutter.dev/flutter/package-collection_collection/MapEquality/hash.html). Any real performance argument I could make seems like splitting hairs looking at the implementation of that versus [hashAll](https://main-api.flutter.dev/flutter/dart-core/Object/hashAll.html)... But I feel more confident using the method that's built for this.

Reply:
Sounds good to me! Thank you both for the feedback 😄